### PR TITLE
lint: naming-convention

### PR DESCRIPTION
**Pull request by justinmk3**
## Problem

`EXTREMELY_LONG_UPPERCASE_NAMES_POPPING_UP_LIKE_PRAIRIE_DOGS_EVERYWHERE`

## Solution

- Enforce `camelCase` by lint rule.
- TODO: Enforce `lessLongName` by lint rule. https://eslint.org/docs/latest/rules/id-length

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by JadenSimon — APPROVED** [review]
Looks like it's working correctly given the errors.

**Review by jpinkney-aws — APPROVED** [review]
I think there are some cases missing since something like: https://github.com/aws/aws-toolkit-vscode/blob/master/src/stepFunctions/constants/aslFormats.ts isn't showing up, but anything is better than nothing at this point

**Comment by justinmk3** [comment]
> [/aslFormats.ts](https://github.com/aws/aws-toolkit-vscode/blob/master/src/stepFunctions/constants/aslFormats.ts) isn't showing up,

thanks for mentioning that.. looking.

**Comment by justinmk3** [comment]
oh actuall the full build log looks ok. it's just github's annotations that are partial (probably to avoid flooding the UI).

File: .eslintrc.js [hunk 1]
```diff
@@ -19,6 +19,27 @@ module.exports = {
...
+                leadingUnderscore: 'allowSingleOrDouble',
+                trailingUnderscore: 'allowSingleOrDouble',
+            },
+            // Allow object properties, enums, and methods to have any format:
+            //   {
+            //     'foo-1-2-3': 42,
+            //   }
+            // https://github.com/typescript-eslint/typescript-eslint/issues/1483#issuecomment-733421303
+            {
+                selector: ['objectLiteralProperty', 'classMethod', 'enumMember'],
```
Comment (justinmk3) [inline .eslintrc.js:38]: could remove "classMethod" here if we change the wizard states

File: src/eventSchemas/templates/schemasAppTemplateUtils.ts [hunk 2]
```diff
@@ -28,31 +28,31 @@ export interface SchemaTemplateParameters {
 
 // This matches the extra_content parameters to Schema-based templates in both key and value names in cookiecutter.json in the templates used by SamEventBridgeSchemaAppPython
 export interface SchemaTemplateExtraContext {
-    AWS_Schema_registry: string
-    AWS_Schema_name: string
-    AWS_Schema_root: string
-    AWS_Schema_source: string
-    AWS_Schema_detail_type: string
-    user_agent: string
+    AWS_Schema_registry: string // eslint-disable-line @typescript-eslint/naming-convention
```
Comment (justinmk3) [inline src/eventSchemas/templates/schemasAppTemplateUtils.ts:31]: Assuming for now that there is a technical need for how these members are spelled.

**Pull request merged**